source/extensions/filters/network/dubbo_proxy/active_message.cc

```diff
@@ -195,9 +196,17 @@ ActiveMessage::~ActiveMessage() {
   parent_.stats().request_active_.dec();
   request_timer_->complete();
   for (auto& filter : decoder_filters_) {
+    ENVOY_LOG(debug, "destroy decoder filter");
     filter->handler()->onDestroy();
   }
-  ENVOY_LOG(debug, "ActiveMessage::~ActiveMessage()");
+
+  for (auto& filter : encoder_filters_) {
+    // Do not call on destroy twice for dual registered filters.
+    if (!filter->dual_filter()) {
```

Comment:
You should be able to use `dual_filter_` directly since this is a friend class?

Reply:
Unlike the http class structure, it is not a friend class.